File: project/static/filial/js/filial.js
function sendData() {
    const url = 'add_filial';  // Flask route URL
    const name = document.getElementById('nomeFilialNova').value;
    const address = document.getElementById('enderecoFilialNova').value;
    const cep = document.getElementById('cepFilialNova').value;
    const nAlunos = document.getElementById('numeroDeAlunosFilialNova').value;
    const myButton = document.getElementById('createModal');
    myButton.disabled = true;
    const data = {
        name: name,
        address: address,
        cep: cep,
        nAlunos: nAlunos
    };

    fetch(url, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(data),
    })
    .then(response => response.json())
    .then(data => {
        console.log('Response from Flask:', data.message);
        myButton.disabled = false;
    })
    .catch(error => {
        console.error('Error sending data:', error);
        myButton.disabled = false;
    })
    .finally(()=>{
        location.reload()
    });    
}

File: project/blueprints/filial/filial.py
from flask import Blueprint,render_template,redirect, request, flash, url_for, jsonify
from project.blueprints.filial.filialService import get_filiais,update_filias, add_filiais, delete_filiais, update_filias
from flask_login import current_user, login_required
from project.utils import adm_required



filial = Blueprint("filial",__name__,url_prefix= '/filial')

@filial.route("/")

@adm_required
def pagina_filiais_route():
    filiais = get_filiais()
    return render_template("filial/filial.html", current_user=current_user, filiais = filiais)

@adm_required
@filial.route("/add_filial", methods=['POST'])
def add_filiais_route():
    data = request.form  # Get the JSON data sent from the client
    
    result = add_filiais(nome=data["nomeFilialNova"], endereco=data["enderecoFilialNova"], cep=data["cepFilialNova"], nAlunos=data["numeroDeAlunosFilialNova"])

    if(result["success"] == 1):
        flash(result["message"], "success")

    else:
        flash(result["message"], "danger")
    
    return redirect(url_for('.pagina_filiais_route'))

@adm_required
@filial.route("/delete_filiais",  methods=['POST'])
def delete_filiais_route():
    data = request.form  # Get the JSON data sent from the client
    
    result = delete_filiais(codigo=data["codigo"])

    if(result["success"] == 1):
        flash(result["message"], "success")

    else:
        flash(result["message"], "danger")
    
    return redirect(url_for('.pagina_filiais_route'))

@adm_required
@filial.route("/update_filiais",  methods=['POST'])
def update_filiais_route():
    data = request.form  # Get the JSON data sent from the client
    result = update_filias(data)

    if(result["success"] == 1):
        flash(result["message"], "success")

    else:
        flash(result["message"], "danger")
    
    return redirect(url_for('.pagina_filiais_route'))